### PR TITLE
Update UFL element interface: move elements to FInAT

### PR DESCRIPTION
**Pull request by mscroggs**
This is an extension of https://github.com/firedrakeproject/tsfc/pull/300.

In https://github.com/FInAT/FInAT/pull/111, I moved the elements from `ufl.legacy` to `finat.ufl`. This PR updates tsfc to use the elements from there.

This is all a consequence of https://github.com/FEniCS/ufl/pull/197.

This PR is coupled with https://github.com/firedrakeproject/firedrake/pull/3166, https://github.com/FInAT/FInAT/pull/111, and https://github.com/NGSolve/ngsPETSc/pull/7.

### CODE REVIEW

File: requirements-git.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-git+https://github.com/firedrakeproject/ufl.git#egg=fenics-ufl
+git+https://github.com/FEniCS/ufl.git
```
Comment (dham) [inline requirements-git.txt:1]: This seems wrong.

**Review by dham — COMMENTED** [review]
fix the imports and this is good to go.

File: requirements-git.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
 git+https://github.com/firedrakeproject/fiat.git#egg=fenics-fiat
-git+https://github.com/FInAT/FInAT.git#egg=finat
+git+https://github.com/FInAT/FInAT.git@mscroggs/ufl-elements
```
Comment (mscroggs) [inline requirements-git.txt:3]: This branch name will need removing once all the branches are merged

**Review by connorjward — DISMISSED** [review]
This is good.

File: requirements-git.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-git+https://github.com/firedrakeproject/ufl.git#egg=fenics-ufl
+git+https://github.com/firedrakeproject/ufl.git@dolci/merge_upstream_2
```
Comment (JDBetteridge) [inline requirements-git.txt:1]:
```suggestion
git+https://github.com/firedrakeproject/ufl.git#egg=fenics-ufl
```

File: requirements-git.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
...
 git+https://github.com/firedrakeproject/fiat.git#egg=fenics-fiat
-git+https://github.com/FInAT/FInAT.git#egg=finat
+git+https://github.com/FInAT/FInAT.git@mscroggs/ufl-elements
```
Comment (JDBetteridge) [inline requirements-git.txt:3]:
```suggestion
git+https://github.com/FInAT/FInAT.git#egg=finat
```

**Review by JDBetteridge — CHANGES_REQUESTED** [review]
Don't worry about adding these I'll add the suggestions to a batch once I've checked the other branches 🙂

**Review by JDBetteridge — APPROVED** [review]

**Pull request merged**